libs/community/langchain_community/llms/octoai_endpoint.py

```diff
@@ -1,166 +1,119 @@
-from typing import Any, Dict, List, Mapping, Optional
+from typing import Any, Dict
 
-from langchain_core.callbacks import CallbackManagerForLLMRun
-from langchain_core.language_models.llms import LLM
-from langchain_core.pydantic_v1 import Extra, root_validator
-from langchain_core.utils import get_from_dict_or_env
+from langchain_core.pydantic_v1 import Field, SecretStr, root_validator
+from langchain_core.utils import convert_to_secret_str, get_from_dict_or_env
 
-from langchain_community.llms.utils import enforce_stop_tokens
+from langchain_community.llms.openai import BaseOpenAI
+from langchain_community.utils.openai import is_openai_v1
 
+DEFAULT_BASE_URL = "https://text.octoai.run/v1/"
+DEFAULT_MODEL = "codellama-7b-instruct"
 
-class OctoAIEndpoint(LLM):
-    """OctoAI LLM Endpoints.
 
-    OctoAIEndpoint is a class to interact with OctoAI
-     Compute Service large language model endpoints.
+class OctoAIEndpoint(BaseOpenAI):
+    """OctoAI LLM Endpoints - OpenAI compatible.
 
-    To use, you should have the ``octoai`` python package installed, and the
-    environment variable ``OCTOAI_API_TOKEN`` set with your API token, or pass
-    it as a named parameter to the constructor.
+    OctoAIEndpoint is a class to interact with OctoAI Compute Service large
+    language model endpoints.
+
+    To use, you should have the environment variable ``OCTOAI_API_TOKEN`` set
+    with your API token, or pass it as a named parameter to the constructor.
 
     Example:
         .. code-block:: python
 
             from langchain_community.llms.octoai_endpoint  import OctoAIEndpoint
-            OctoAIEndpoint(
-                octoai_api_token="octoai-api-key",
-                endpoint_url="https://text.octoai.run/v1/chat/completions",
-                model_kwargs={
-                    "model": "llama-2-13b-chat-fp16",
-                    "messages": [
-                        {
-                            "role": "system",
-                            "content": "Below is an instruction that describes a task.
-                                Write a response that completes the request."
-                        }
-                    ],
-                    "stream": False,
-                    "max_tokens": 256,
-                    "presence_penalty": 0,
-                    "temperature": 0.1,
-                    "top_p": 0.9
-                }
+
+            llm = OctoAIEndpoint(
+                model="llama-2-13b-chat-fp16",
+                max_tokens=200,
+                presence_penalty=0,
+                temperature=0.1,
+                top_p=0.9,
             )
 
     """
 
-    endpoint_url: Optional[str] = None
-    """Endpoint URL to use."""
-
-    model_kwargs: Optional[dict] = None
-    """Keyword arguments to pass to the model."""
```

Comment:
removing these is a breaking change, could we add some handling for them in the root validator?